### PR TITLE
Release: fix running pre/post release scripts in windows

### PR DESCRIPTION
**Pull request by timmywil**
- also fix tagging the release in the dist repo

### Summary ###

I was attempting the release and ran into some minor issues. I did some more manual dry runs of the pre-release.sh and post-release.sh scripts and this *should* work now.

### Checklist ###

* [x] New tests have been added to show the fix or feature works
* [x] If needed, a docs issue/PR was created at https://github.com/jquery/api.jquery.com

### CODE REVIEW

File: build/release/post-release.sh [hunk 1]
```diff
@@ -33,7 +33,9 @@ cd -
 npm run release:dist $1 $2
 cd $dist
 git add -A
-npm version $1
+git commit -m "Release: $1"
+# -s to sign and annotate tag (recommended for releases)
+git tag -s $1 -m "Release: $1"
```
Comment (timmywil) [inline build/release/post-release.sh:38]: `npm version` wasn't quite right. Aside from the need to commit first anyway, the correct version is already set at this point in the package.json. But more importantly, there's no `.npmrc` in the jquery-dist repo, so the tag name would be wrong.

Comment (mgol) [reply build/release/post-release.sh:38]: The correct version is needed not just in `package.json`, but also in the lockfile - and there, depending on the version, even twice! For this reason, it’s be safer to rely on npm version; the format can be tweaked, I believe.

Comment (timmywil) [reply build/release/post-release.sh:38]: This is for jquery-dist. There's no lockfile.

Comment (mgol) [reply build/release/post-release.sh:38]: Ah, right. There are no dependencies there so it wouldn’t be useful to commit one.

**Review by mgol — APPROVED** [review]

**Pull request merged**